### PR TITLE
refactor sensitive check into a new moderation service

### PR DESCRIPTION
**Pull request by Rader**
- new moderation service to replace sensitive checking
- introduce Temporal workflow to run repo content sensitive checking
- add DFA based local sensitive words detecting

### MR Summary:
*The summary is added by @codegpt.*

The Merge Request introduces a new moderation service to handle sensitive content checking in repository content. It replaces the existing sensitive checking mechanism with a more structured approach, including the integration of Temporal workflow for asynchronous processing and a DFA (Deterministic Finite Automaton) based local sensitive words detection mechanism. Key updates include:

1. Implementation of a new moderation service for sensitive content checking.
2. Integration of Temporal workflow for asynchronous processing of repo content checks.
3. Introduction of a DFA based mechanism for detecting sensitive words locally.
4. Refactoring of existing code to utilize the new moderation service across various components such as user, organization, and repository handling.
5. Addition of new tests and modification of existing ones to align with the new moderation service.
6. Creation of new handlers, components, and workflow activities to support the moderation service functionalities.
7. Update of documentation and Swagger API definitions to reflect the changes introduced by the moderation service.

### CODE REVIEW

**Review by ganisback — APPROVED** [review]
LGTM

File: moderation/component/repo_file_content_reader.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+			Path:      c.file.Path,
+			RepoType:  c.file.Repository.RepositoryType,
+			Ref:       c.file.Repository.DefaultBranch,
+		}
+
+		ctx := context.Background()
+		var err error
+		c.innerReader, _, err = c.git.GetRepoFileReader(ctx, req)
+		if err != nil {
+			slog.Error("failed to create git file reader", slog.Any("error", err), slog.String("path", c.file.Path), slog.Int64("repository_file_id", c.file.ID))
```
[the remark this answers is not in the document]
Comment (Rader) [reply moderation/component/repo_file_content_reader.go:62]: there is no `errors` package

File: api/handler/model.go [hunk 2]
```diff
@@ -22,7 +22,10 @@ func NewModelHandler(config *config.Config) (*ModelHandler, error) {
 	if err != nil {
 		return nil, err
 	}
-	sc := component.NewSensitiveComponent(config)
+	sc, err := component.NewSensitiveComponent(config)
+	if err != nil {
+		return nil, fmt.Errorf("error creating sensitive component:%w", err)
```
[the remark this answers is not in the document]
Comment (Rader) [reply api/handler/model.go:27]: there is no `errors` package

File: api/router/api.go [hunk 3]
```diff
@@ -240,7 +240,10 @@ func NewRouter(config *config.Config, enableSwagger bool) (*gin.Engine, error) {
 	apiGroup.GET("/callback/casdoor", userProxyHandler.Proxy)
 	// Sensive check
 	if config.SensitiveCheck.Enable {
-		sensitiveCtrl := handler.NewSensitiveHandler(config)
+		sensitiveCtrl, err := handler.NewSensitiveHandler(config)
```
[the remark this answers is not in the document]
Comment (Rader) [reply api/router/api.go:243]: error logged in the next line

**Pull request merged**